### ISSUE TITLE
`numba` automatic type inference failure in `calc_distance_coords_pbc`

### ISSUE THREAD

**Issue by w-k-jones** [tobac-project/tobac#383]
In some circumstances, `numba` can raise the following exception in `calc_distance_coords_pbc`:
```
TypingError: Failed in nopython mode pipeline (step: nopython frontend)
No implementation of function Function(<built-in function sub>) found for signature:
 
 >>> sub(none, none)
 
There are 14 candidate implementations:
  - Of which 10 did not match due to:
  Overload of function 'sub': File: <numerous>: Line N/A.
    With argument(s): '(none, none)':
   No match.
  - Of which 2 did not match due to:
  Operator Overload in function 'sub': File: unknown: Line unknown.
    With argument(s): '(none, none)':
   No match for registered cases:
    * (int64, int64) -> int64
    * (int64, uint64) -> int64
    * (uint64, int64) -> int64
    * (uint64, uint64) -> uint64
    * (float32, float32) -> float32
    * (float64, float64) -> float64
    * (complex64, complex64) -> complex64
    * (complex128, complex128) -> complex128
  - Of which 2 did not match due to:
  Overload in function 'impl_set_difference': File: numba/cpython/setobj.py: Line 1525.
    With argument(s): '(none, none)':
   Rejected as the implementation raised a specific error:
     TypingError: All arguments must be Sets, got (none, none)
  raised from [/Users/jonesw/mambaforge-pypy3/envs/tobac_v1.5/lib/python3.11/site-packages/numba/cpython/setobj.py:108](https://file+.vscode-resource.vscode-cdn.net/Users/jonesw/mambaforge-pypy3/envs/tobac_v1.5/lib/python3.11/site-packages/numba/cpython/setobj.py:108)

During: typing of intrinsic-call at [/Users/jonesw/python/tobac/tobac/utils/periodic_boundaries.py](https://file+.vscode-resource.vscode-cdn.net/Users/jonesw/python/tobac/tobac/utils/periodic_boundaries.py) (243)

File "../../tobac/tobac/utils/periodic_boundaries.py", line 243:
def calc_distance_coords_pbc(
    <source elided>
    if PBC_flag in ["hdim_1", "both"]:
        size_h1 = max_h1 - min_h1
        ^
```
This can be fixed by supplying `njit` with a function signature, but this requires some changes to how we apply the decorator. I think this should be added to the v1.5.2 release

Also `numba` is not listed as a requirement, so is not tested in CI at present. We should look into how to implement this...

**Comment by w-k-jones** [comment]
I think this may be linked to #332

**Comment by w-k-jones** [comment]
Ok I have zero clue what is going on here. I tested this again on a separate system. 5 days ago PBC tracking ran fine. Today I get the same error. There have been no changes to either tobac or the conda environment in that time.

**Comment by JuliaKukulies** [comment]
Thank you for fixing this so quickly!

**Comment by w-k-jones** [comment]
Resolved with #384
